oa4mp/serve.go

```diff
@@ -306,6 +306,27 @@ func ConfigureOA4MP() (launcher daemon.Launcher, err error) {
 		return
 	}
 
+	// Ensure the OA4MP storage directory exists and has correct permissions and ownership
+	OA4MPStoragePath := "/opt/scitokens-server/var/storage"
+
+	if err := os.Chmod(OA4MPStoragePath, 0700); err != nil {
+		if os.IsNotExist(err) {
+			log.Debugln("OA4MP storage directory does not exist. Creating", OA4MPStoragePath)
+			if err = os.MkdirAll(OA4MPStoragePath, 0755); err != nil {
+				log.Fatal("Failed to create directory", OA4MPStoragePath, ":", err)
```

Comment:
Let's avoid `log.Fatal`, which terminates the program by calling `os.Exit(1)`. Instead, when there's an error, it should be returned. Notice the rest of this function does something like this when it encounters an error:
```
if err != nil {
		err = errors.Wrap(err, "Failed to load the private issuer key for running issuer")
		return
	}
```
Because the names of the function's return variables are declared in the function signature, this return is actually returning the `err` variable that's set in the conditional.

That is, rather than log the error, use `errors.Wrap` or `errors.Wrapf` to set `err` and then return.

Reply:
Thanks for your reviews! I'm starting to address these issues you mentioned